File: system-test/dotnet/Dsl.Core/UseCase/External/Erp/UseCases/GetProductVerification.cs
using Driver.Port.External.Erp.Dtos;
using Dsl.Core.Shared;
using Shouldly;

namespace Dsl.Core.External.Erp.UseCases;

public class GetProductVerification : ResponseVerification<GetProductResponse>
{
    public GetProductVerification(GetProductResponse response, UseCaseContext context)
        : base(response, context)
    {
    }

    public GetProductVerification Sku(string skuParamAlias)
    {
        var expectedSku = Context.GetParamValue(skuParamAlias);
        var actualSku = Response.Sku;
        actualSku.ShouldBe(expectedSku, $"Expected SKU to be '{expectedSku}', but was '{actualSku}'");
        return this;
    }

    public GetProductVerification Price(decimal expectedPrice)
    {
        var actualPrice = Response.Price;
        actualPrice.ShouldBe(expectedPrice, $"Expected price to be {expectedPrice}, but was {actualPrice}");
        return this;
    }

    public GetProductVerification Price(string expectedPrice)
    {
        return Price(decimal.Parse(expectedPrice));
    }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/core/usecase/external/erp/usecases/GetProductVerification.java
package com.optivem.shop.dsl.core.usecase.external.erp.usecases;

import com.optivem.shop.dsl.driver.port.external.erp.dtos.GetProductResponse;
import com.optivem.shop.dsl.common.Converter;
import com.optivem.shop.dsl.core.shared.ResponseVerification;
import com.optivem.shop.dsl.core.shared.UseCaseContext;

import java.math.BigDecimal;

import static org.assertj.core.api.Assertions.assertThat;

public class GetProductVerification extends ResponseVerification<GetProductResponse> {
    public GetProductVerification(GetProductResponse response, UseCaseContext context) {
        super(response, context);
    }

    public GetProductVerification sku(String skuParamAlias) {
        var expectedSku = getContext().getParamValue(skuParamAlias);
        var actualSku = getResponse().getSku();
        assertThat(actualSku)
                .withFailMessage("Expected SKU to be '%s', but was '%s'", expectedSku, actualSku)
                .isEqualTo(expectedSku);
        return this;
    }

    public GetProductVerification price(BigDecimal expectedPrice) {
        var actualPrice = getResponse().getPrice();
        assertThat(actualPrice)
                .withFailMessage("Expected price to be %s, but was %s", expectedPrice, actualPrice)
                .isEqualByComparingTo(expectedPrice);
        return this;
    }

    public GetProductVerification price(double expectedPrice) {
        return price(Converter.toBigDecimal(expectedPrice));
    }

    public GetProductVerification price(String expectedPrice) {
        return price(Converter.toBigDecimal(expectedPrice));
    }
}
